### ISSUE TITLE
Segment.io Compatible Batch API Endpoint

### ISSUE THREAD

**Issue by jspizziri** [PostHog/posthog#24674]
## Is your feature request related to a problem?

Not related to a problem, but it would be great if there was a segment.io compatible batch endpoint that mapped events into the native Posthog batch format: https://posthog.com/docs/api/capture

If this existed, then  one could integrate with Posthog Analytics with any of the preexisting segment.io (and segment maintained) client SDKs, like [this one for example](https://github.com/segmentio/analytics-react-native). For customers who are already on Segment, or are migrating away from Segment, it would make it really easy to migrate. Additionally, it could potentially reduce the overall surface area of the software that Posthog is required to maintain.

## Describe the solution you'd like

Basically, I'd want to be able to point segment analytics libraries at a `/batch` endpoint hosted by Posthog. Posthog would process the events and manage the mapping of them into the Posthog schema.

## Describe alternatives you've considered

Migrating to [the Posthog react-native library](https://posthog.com/docs/libraries/react-native), but I already have a segment integration, and I'm lazy :D.

### Debug info

```shell
Location: https://us.posthog.com/project/73088/data-warehouse
Session: https://us.posthog.com/project/sTMFPsFhdP1Ssg/replay/01919df7-0b21-7535-a73a-840642364626?t=3217
Admin: http://go/adminOrgUS/018fef23-3ebd-0000-f5f4-facdd7e687e6 (project ID 73088)
Sentry: http://go/sentryUS/73088
```
